src/tests/fuchsia.md

```diff
@@ -0,0 +1,162 @@
+build takes 46G of space – and as you might imagine, it takes awhile to
+complete.
+
+### Modifying the Fuchsia checkout
+
+The main reason you would want to build Fuchsia locally is because you need to
+investigate a regression. After running a Docker build, you'll find the Fuchsia
+checkout inside the `obj/fuchsia` directory of your Rust checkout.  If you
+modify the `KEEP_CHECKOUT` line in the [build-fuchsia.sh] script to
+`KEEP_CHECKOUT=1`, you can change the checkout as needed and rerun the build
+command above. This will reuse all the build results from before.
+
+You can find more options to customize the Fuchsia checkout in the
+[build-fuchsia.sh] script.
+
+### Customizing the Fuchsia build
+
+You can find more info about the options used to build Fuchsia in Rust CI in the
+[build_fuchsia_from_rust_ci.sh] script invoked by [build-fuchsia.sh].
+
+The Fuchsia build system uses [GN], a metabuild system that generates [Ninja]
+files and then hands off the work of running the build to Ninja.
+
+Fuchsia developers use `fx` to run builds and perform other development tasks.
+This tool is located in `.jiri_root/bin` of the Fuchsia checkout; you may need
+to add this to your `$PATH` for some workflows.
+
+There are a few `fx` subcommands that are relevant, including:
+
+* `fx set` accepts build arguments, writes them to `out/default/args.gn`, and runs GN.
+* `fx build` builds the Fuchsia project using Ninja. It will automatically pick
+  up changes to build arguments and rerun GN. By default it builds everything,
+  but it also accepts target paths to build specific targets (see below).
+* `fx clippy` runs Clippy on specific Rust targets (or all of them). We use this
+  in the Rust CI build to avoid running codegen on most Rust targets. Underneath
+  it invokes Ninja, just like `fx build`. The clippy results are saved in json
+  files inside the build output directory before being printed.
+
+#### Target paths
+
+GN uses paths like the following to identify build targets:
+
+```
+//src/starnix/kernel:starnix_core
+```
+
+The initial `//` means the root of the checkout, and the remaining slashes are
+directory names. The string after `:` is the _target name_ of a target defined
+in the `BUILD.gn` file of that directory.
+
+The target name can be omitted if it is the same as the directory name. In other
+words, `//src/starnix/kernel` is the same as `//src/starnix/kernel:kernel`.
+
+These target paths are used inside `BUILD.gn` files to reference dependencies,
+and can also be used in `fx build`.
+
+#### Modifying compiler flags
+
+You can put custom compiler flags inside a GN `config` that is added to a target.
+As a simple example:
+
+```
+config("everybody_loops") {
+    rustflags = [ "-Zeverybody-loops" ]
+}
+
+rustc_binary("example") {
+    crate_root = "src/bin.rs"
+    # ...existing keys here...
+    configs += [ ":everybody_loops" ]
+}
+```
+
+This will add the flag `-Zeverybody-loops` to rustc for the given target. You can
```

Comment:
could we explicitly mention the target name here just to make it super clear? (I don't expect people to read a lot of GN docs, so it will be nice that we have an example here)